vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -272,25 +272,34 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 	filters.Add("status", "running")
 	filters.Add("status", "paused")
 	filters.Add("status", "restarting")
-	containers, err := c.ContainerList(context.Background(), types.ContainerListOptions{
+
+	ctx, cancel := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
+	defer cancel()
+	containers, err := c.ContainerList(ctx, types.ContainerListOptions{
 		All:    true,
 		Filter: filters,
 	})
 	if err != nil {
+		log.Errorf("ContainerList failed (err: %v)", err)
 		return err
 	}
 
 	log.Debugf("Found %d running or paused containers", len(containers))
 	for _, ct := range containers {
-		containerJSONInfo, err := c.ContainerInspect(context.Background(), ct.ID)
+		ctx_inspect, cancel_inspect := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
+		defer cancel_inspect()
```

Comment:
Ok, my doubt was the reassignment of cancel_inspect var. Will the defer statement ensure that the cancel function is called for each created context or just the last one? Looks like defer statement evaluates the args when the statement is executed and saves the defer'ed function to call. It should be ok then.